enterprise/reporting/error_extractor.go

```diff
@@ -31,21 +32,52 @@ var (
 	whitespacesRegex = regexp.MustCompile("[ \t\n\r]*") // used in checking if string is a valid json to remove extra-spaces
 
 	defaultErrorMessageKeys = []string{"message", "description", "detail", "title", errorKey, "error_message"}
+	deprecationKeywords     = map[string]int{
+		"deprecated":               2,
+		"deprecation":              2,
+		"version":                  1,
+		"obsolete":                 1,
+		"outdated":                 1,
+		"end of life":              4,
+		"legacy":                   1,
+		"discontinued":             1,
+		"retired":                  2,
+		"no longer supported":      3,
+		"old version":              1,
+		"deprecated software":      2,
+		"upgrade required":         2,
+		"obsolete version":         1,
+		"unsupported version":      2,
+		"deprecated feature":       2,
+		"version no longer valid":  2,
+		"deprecated library":       2,
+		"version upgrade":          2,
+		"deprecated component":     2,
+		"upgrade recommended":      2,
+		"end-of-support":           3,
+		"discontinued product":     1,
+		"deprecated functionality": 2,
+		"version obsolescence":     1,
+		"deprecated module":        2,
+	}
 )
 
 type ExtractorHandle struct {
-	log              logger.Logger
-	ErrorMessageKeys []string // the keys where in we may have error message
+	log                              logger.Logger
+	ErrorMessageKeys                 []string // the keys where in we may have error message
+	versionDeprecationThresholdScore misc.ValueLoader[int]
 }
 
 func NewErrorDetailExtractor(log logger.Logger) *ExtractorHandle {
 	errMsgKeys := config.GetStringSlice("Reporting.ErrorDetail.ErrorMessageKeys", []string{})
+	versionDepThreshold := config.GetReloadableIntVar(1, 1, "Reporting.ErrorDetail.versionDeprecationThresholdScore")
```

Comment:
should we keep min score to be 2 as version keep might come in different context also

Reply:
This is a reloadable configuration variable. We can put the desired value in operator